### DISCUSSION TITLE
The 'credentials' field on 'RequestInitializerDict' is not implemented.

### DISCUSSION THREAD

**Discussion by tarikrital** [swellstores/community#465] [Help] [answered]
I'm using Astro build project using SSR and it works ok locally but when I use wrangler to prepare for Cloudflare it gives the following error:

`The 'credentials' field on 'RequestInitializerDict' is not implemented.`

The important is the issue is when using `swell-js` it gives this error but when using `fetch` and a public endpoint it works both locally and in wrangler.

**Comment by logeshswell** [comment]
Looks like this error doesn't belong to the swell-js implementation.

Do you mean that `swell-js` package is not installed in cloudflare?

Please check if the below information from Wrangler docs helps you.
https://developers.cloudflare.com/workers/wrangler/bundling/

**Reply by tarikrital** [reply]
I see the first issue is Wrangler locally why a normal fetch to public API works but Swell fetch doesn't! (line before the error)

```
const categories = await swell.categories.list({
  limit: 100,
  page: 1,
});
```

> The 'credentials' field on 'RequestInitializerDict' is not implemented.

**Reply by tarikrital** [reply]
Can I know why **closing** the issue, I work on a project of your client!

**Reply by logeshswell** [reply]
> why a normal fetch to public API works but Swell fetch doesn't!

Do you mean that Swell fetch (eg: `await swell.categories.list `) doesn't work locally?

**Reply by tarikrital** [reply]
Yes it works fine using `pnpm dev` but when doing a preview locally it gives 500 error with a log on the console as I mentioned.

`The 'credentials' field on 'RequestInitializerDict' is not implemented.`

>       at checkURL
>   (file:///home/wrangler/tmp/bundle-ngYvmD/checked-fetch.js:9:9)
>       at Object.apply
>   (file:///home/.wrangler/tmp/bundle-ngYvmD/checked-fetch.js:27:3)
>       at e (file:///home/dist/_worker.js:49:109124)
>       at Object.list (file:///home/dist/_worker.js:49:41677)
>       at e (file:///home/dist/_worker.js:57:1116)
>       at index (file:///home/dist/_worker.js:9:222)
>       at im (file:///home/zdist/_worker.js:23:284)
>       at P_ (file:///home/dist/_worker.js:22:119)
>       at am (file:///home/dist/_worker.js:38:2579)
>       at xx (file:///home/dist/_worker.js:108:15205) (x6)
> 
> 
> [wrangler:inf] GET / 500 Internal Server Error (32ms)
>

**Reply by logeshswell** [reply]
I doubt that `swell-js` package is not installed in Cloudflare, while `fetch` package might be installed by default.
Please confirm the same with Cloudflare team. 

Also, follow this documentation to install npm packages into your cloudflare worker:
https://developers.cloudflare.com/workers/wrangler/bundling/

**Reply by tarikrital** [reply]
I'm not sure why you selected the answer as solution and I the reporter of this issue didn't find the solution yet!

I see before publishing the project to Cloudflare worker it should work locally first and that is the benefit of Wrangler.

> wrangler dev is a way to [locally test](https://developers.cloudflare.com/workers/observability/local-development-and-testing/) your Worker while developing. With wrangler dev running, send HTTP requests to localhost:8787 and your Worker should execute as expected. You will also see console.log messages and exceptions appearing in your terminal.

**Reply by logeshswell** [reply]
I get you, but did you try installing swell-js in Wrangler dev?
Can you check whether swell-js library present in node_modules dir?

**Reply by logeshswell** [reply]
I am still saying that this error doesn't belong to swell-js. So, I couldn't guide you properly.

**Reply by tarikrital** [reply, marked as answer]
The library exists that is the reason why it works locally!

We switched to Vercel, I suggest to increase the support for Swell clients...
